x-pack/plugins/uptime/common/runtime_types/synthetics_service_api_key.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+
+export const SyntheticsServiceApiKeyType = t.type({
+  id: t.string,
+  name: t.string,
+  apiKey: t.string,
```

Comment:
In practice, I've also seen the api service return an `encoded` key on the object, which is the encoded id and name which saves us the trouble of having to encode it ourselves.

Reply:
i think we are not using the encoded part, since we only need apiKey and that will get saved in savedObjects in encrypted form. But will add it just in case for typing purpose.

Reply:
we are not saving `encoded` part at all in the saved object